examples/quickstart/jupyter-notebooks/notebooks/03-query/04-UnionOperations.ipynb

```diff
@@ -0,0 +1,644 @@
+    "  \"flags\",\n",
+    "  \"isUnpatrolled\",\n",
+    "  \"page\",\n",
+    "  \"diffUrl\",\n",
+    "  \"added\",\n",
+    "  \"comment\",\n",
+    "  \"commentLength\",\n",
+    "  \"isNew\",\n",
+    "  \"isMinor\",\n",
+    "  \"delta\",\n",
+    "  \"isAnonymous\",\n",
+    "  \"user\",\n",
+    "  \"deltaBucket\",\n",
+    "  \"deleted\",\n",
+    "  \"namespace\",\n",
+    "  \"cityName\",\n",
+    "  \"countryName\",\n",
+    "  \"regionIsoCode\",\n",
+    "  \"metroCode\",\n",
+    "  \"countryIsoCode\",\n",
+    "  \"regionName\"\n",
+    "FROM \"ext\"\n",
+    "WHERE \"channel\" LIKE '#fr%'\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "sql_client.run_task(sql)\n",
+    "sql_client.wait_until_ready('wikipedia-fr')\n",
+    "display.table('wikipedia-fr')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8bbf2c6-681a-46f5-82f2-201cbbe8058d",
+   "metadata": {},
+   "source": [
+    "In the next cell, we declare our `UNION ALL` data source as the single, unified view of all the data. Then we use that in our outer `SELECT` to count the number of robot and non-robot edits by channel.\n",
+    "\n",
+    "Remember, the `SELECT` in the `unifiedSource` must be simple in order to meet the constraints set by a table level `UNION ALL`, so any filtering ans so on can only be done in the outer `SELECT` statement."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "783fe77d-2e7b-476a-9748-67ea90c8bb91",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "WITH unifiedSource AS (\n",
+    "    SELECT\n",
+    "        \"__time\",\n",
+    "        \"isRobot\",\n",
+    "        \"channel\",\n",
+    "        \"user\",\n",
+    "        \"countryName\"\n",
+    "    FROM \"wikipedia-en\"\n",
+    "    UNION ALL\n",
+    "    SELECT\n",
+    "        \"__time\",\n",
+    "        \"isRobot\",\n",
+    "        \"channel\",\n",
+    "        \"user\",\n",
+    "        \"countryName\"\n",
+    "    FROM \"wikipedia-fr\"\n",
+    "    )\n",
+    "\n",
+    "SELECT\n",
+    "    \"channel\",\n",
+    "    COUNT(*) FILTER (WHERE isRobot=true) AS \"Robot Edits\",\n",
+    "    COUNT (DISTINCT user) FILTER (WHERE isRobot=true) AS \"Robot Editors\",\n",
+    "    COUNT(*) FILTER (WHERE isRobot=false) AS \"Human Edits\",\n",
+    "    COUNT (DISTINCT user) FILTER (WHERE isRobot=false) AS \"Human Editors\"\n",
+    "FROM unifiedSource\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f58a1846-5072-4495-b840-a620de3c0442",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "* There are two modes for `UNION ALL` in Druid - top level and table level\n",
+    "* Top level is a simple concatenation, and operations must be done on the source `TABLE`s\n",
+    "* Table level uses a `union` data source, and operations must be done on the outer `SELECT`\n",
```

Comment:
I really like this conclusion block. I was confused by some of the later examples, so these points weren't very clear to me before this. Conceptually, you can add the following to make it easier to follow:
* Include a table of contents section and more descriptive section headers. Right now we have the following:
   * **Concatenating result sets with UNION ALL** --> consider referring to this as "Top-level"
   * **Working with concatenated result sets** --> consider referring to this as "Table-level"
* Add a bit more details to the introductory sentence. For example, "This notebook introduces the UNION ALL operator and shows two contexts for how this operator can be used: in top-level queries and in table-level queries."
* Add more description to table-level queries. Top-level ones seem to be more intuitive, but it could help to have more details around what table-level queries are, after this sentence (and before jumping into the example)
> Instead of a top level UNION ALL, it switches to a "[table level](https://druid.apache.org/docs/26.0.0/querying/sql.html#table-level)" query.

Reply:
Thank you! I ended up removing the trap that generates the error condition, and pulling a lot back up under the main headings.